R/pkg/R/functions.R

```diff
@@ -2328,22 +2343,27 @@ setMethod("schema_of_json", signature(x = "characterOrColumn"),
 #' If the string is unparseable, the Column will contain the value NA.
 #'
 #' @rdname column_collection_functions
-#' @aliases from_csv from_csv,Column,character-method
+#' @aliases from_csv from_csv,Column,characterOrstructTypeOrColumn-method
 #' @examples
 #'
 #' \dontrun{
-#' df <- sql("SELECT 'Amsterdam,2018' as csv")
+#' csv <- "Amsterdam,2018"
+#' df <- sql(paste0("SELECT '", csv, "' as csv"))
 #' schema <- "city STRING, year INT"
-#' head(select(df, from_csv(df$csv, schema)))}
+#' head(select(df, from_csv(df$csv, schema)))
+#' head(select(df, from_csv(df$csv, structType(schema))))
+#' head(select(df, from_csv(df$csv, schema_of_csv(csv))))}
 #' @note from_csv since 3.0.0
-setMethod("from_csv", signature(x = "Column", schema = "characterOrColumn"),
+setMethod("from_csv", signature(x = "Column", schema = "characterOrstructTypeOrColumn"),
           function(x, schema, ...) {
-            if (class(schema) == "Column") {
-              jschema <- schema@jc
-            } else if (is.character(schema)) {
+            if (class(schema) == "structType") {
+              schema <- callJMethod(schema$job, "toDDL")
```

Comment:
`schema$jobj`? is there test for this?

Reply:
I don't quite get this - is it because from_csv doesn't take StructType?

Reply:
Hm. Weird, this is being tested below `from_csv(df$col, structType("a INT"))`.

Reply:
Yup, I did this because `from_csv` with Java map doesn't take StructType ..

https://github.com/apache/spark/blob/7c7fccfeb5bc079fede41eb64f57ab6b1b4b9018/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L3790

Reply:
R has some funky partial/prefix matching

Reply:
Haha. I haven't been aware of that so far!